src/context.rs

```diff
@@ -234,3 +245,186 @@ impl PendingFix {
+                    }
+                    loop_landed_on_comment_or_node!(self);
+                }
+                ReturnedCurrentNode => {
+                    move_to_next_sibling_or_go_to_parent_and_loop!(self);
+                    loop_landed_on_comment_or_node!(self);
+                }
+                OnComment => {
+                    move_to_next_sibling_or_go_to_parent_and_loop!(self);
+                    loop_landed_on_comment_or_node!(self);
+                }
+                LandedOnNonCommentNode => {
+                    if !self.cursor.goto_first_child() {
+                        self.state = ReturnedCurrentNode;
+                        return Some(self.cursor.node());
+                    }
+                    loop_landed_on_comment_or_node!(self);
+                }
+                JustReturnedToParent => {
+                    if self.cursor.node() == self.original_node {
+                        loop_done!(self);
+                    }
+                    move_to_next_sibling_or_go_to_parent_and_loop!(self);
+                    loop_landed_on_comment_or_node!(self);
+                }
+            }
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+enum TokenWalkerState {
+    Initial,
+    OnComment,
+    ReturnedCurrentNode,
+    JustReturnedToParent,
+    LandedOnNonCommentNode,
+    Done,
+}
+
+#[cfg(test)]
+mod tests {
+    use tree_sitter_grep::tree_sitter::Parser;
+
+    use super::*;
+
+    fn test_all_tokens_text(text: &str, all_tokens_text: &[&str]) {
+        let mut parser = Parser::new();
+        parser
+            .set_language(SupportedLanguage::Javascript.language())
+            .unwrap();
+        let tree = parser.parse(text, None).unwrap();
+        assert_eq!(
+            get_tokens(tree.root_node())
+                .map(|node| node.utf8_text(text.as_bytes()).unwrap())
+                .collect::<Vec<_>>(),
+            all_tokens_text
+        );
+    }
+
+    #[test]
+    fn test_get_tokens_simple() {
+        test_all_tokens_text("const x = 5;", &["const", "x", "=", "5", ";"]);
+    }
+
+    #[test]
+    fn test_get_tokens_structured() {
+        test_all_tokens_text(
+            r#"
+            const whee = function(foo) {
+                for (let x = 1; x < 100; x++) {
+                    foo(x);
+                }
+            }
+        "#,
+            &[
+                "const", "whee", "=", "function", "(", "foo", ")", "{", "for", "(", "let", "x",
+                "=", "1", ";", "x", "<", "100", ";", "x", "++", ")", "{", "foo", "(", "x", ")",
+                ";", "}", "}",
+            ],
+        );
+    }
+
+    #[test]
+    fn test_get_tokens_omits_comments() {
+        test_all_tokens_text(
+            r#"
+            /*leading*/
+            const x = 5
+            /*hello*/ // whee
+            foo()
+            // trailing
+        "#,
+            &["const", "x", "=", "5", "foo", "(", ")"],
+        );
+    }
```

Comment:
Should add tests for the "special cases" of just a single token and just comment tokens (and now that I think of it, things like just a single token amidst surrounding comment tokens)